Sources/Controllers/2.0/RoutePlanning/OASaveTrackViewController.mm

```diff
@@ -343,6 +330,7 @@ - (nonnull UITableViewCell *) tableView:(nonnull UITableView *)tableView cellFor
         {
             cell.delegate = self;
             [cell setValues:item[@"values"] sizes:nil colors:nil addButtonTitle:item[@"addButtonTitle"] withSelectedIndex:(int)[item[@"selectedValue"] intValue]];
+            [cell.collectionView layoutIfNeeded];
             [cell.collectionView scrollToItemAtIndexPath:[NSIndexPath indexPathForItem:_selectedFolderIndex inSection:0]
```

Comment:
Extract scroll logic to the cell, store the currently selected index in the cell, and scroll to the index only! if the index has changed. Call the new method in willDisplayCellForRow: instead of cellForRow.